### PR TITLE
fix mutiple memory issue

### PR DESCRIPTION
**Pull request by nanli1**
Test result:
 
```

[root@dell-per730-59 tp-libvirt]# rpm -q libvirt
libvirt-9.5.0-0rc1.1.el9.x86_64
[root@dell-per730-59 tp-libvirt]# cat /proc/version
Linux version 5.14.0-333.el9.x86_64 (mockbuild@x86-64-01.build.eng.rdu2.redhat.com) (gcc (GCC) 11.4.1 20230605 (Red Hat 11.4.1-2), GNU ld version 2.35.2-42.el9) #1 SMP PREEMPT_DYNAMIC Wed Jun 28 07:51:51 EDT 2023
 [root@dell-per730-59 tp-libvirt]# avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 memory.devices.lifecycle.mutiple_mem --vt-connect-uri qemu:///system

 (1/1) type_specific.io-github-autotest-libvirt.memory.devices.lifecycle.mutiple_mem: PASS (197.86 s)

```

### CODE REVIEW

**Comment by nanli1** [comment]
```
avocado run --vt-type libvirt --test-runner=runner --vt-machine-type q35 memory.backing.page_locking_and_shared_pages
 (01/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.page_default.ksm_default.lock_without_hard_limit: PASS (55.84 s)
 (02/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.page_default.ksm_default.lock_hard_limit: PASS (54.87 s)
 (03/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.page_default.ksm_disabled.lock_default: PASS (56.71 s)
 (04/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.page_default.ksm_disabled.lock_without_hard_limit: PASS (60.20 s)
 (05/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.page_default.ksm_disabled.lock_hard_limit: PASS (61.16 s)
 (06/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.hugepage.ksm_default.lock_without_hard_limit: PASS (60.68 s)
 (07/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.hugepage.ksm_default.lock_hard_limit: PASS (56.88 s)
 (08/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.hugepage.ksm_disabled.lock_default: PASS (60.80 s)
 (09/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.hugepage.ksm_disabled.lock_without_hard_limit: PASS (55.48 s)
 (10/10) type_specific.io-github-autotest-libvirt.memory.backing.page_locking_and_shared_pages.memory_allocation.hugepage.ksm_disabled.lock_hard_limit: PASS (57.81 s)

```

File: libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py [hunk 1]
```diff
@@ -260,14 +252,36 @@ def get_affected_mem(params, operation):
...
     return memory_affected, current_mem_affected
 
 
+def update_vm_devices(vmxml, params):
+    """
+    Update vm devices.
+
+    :param vmxml: vmxml need to be added memory devices.
+    :param params: Dictionary with the test parameters.
+    :return: vmxml, vmxml which has been added memory devices.
```
Comment (chunfuwen) [inline libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py:269]: return devices, not vmxml

Comment (nanli1) [reply libvirt/tests/src/memory/memory_devices/lifecycle_of_mixed_memory_devices.py]: Thanks chunfu,Done

**Comment by dzhengfy** [comment]
@nanli1 please add what the multiple mem issues are and please remove any internal test machine information as well as libvirt downstream version.

**Review by dzhengfy — APPROVED** [review]
Codes LGTm

**Review by chunfuwen — APPROVED** [review]
lgtm

**Pull request merged**